Firmware/communication/ascii_protocol.cpp

```diff
@@ -36,14 +36,16 @@ static Introspectable root_obj = ODriveTypeInfo<ODrive>::make_introspectable(odr
 // @brief Sends a line on the specified output.
 template<typename ... TArgs>
 void respond(StreamSink& output, bool include_checksum, const char * fmt, TArgs&& ... args) {
-    char response[64];
+    char response[64]; // Hardcoded max buffer size. We silently truncate the output if it's too long for the buffer.
     size_t len = snprintf(response, sizeof(response), fmt, std::forward<TArgs>(args)...);
+    len = std::min(len, sizeof(response));
     output.process_bytes((uint8_t*)response, len, nullptr); // TODO: use process_all instead
     if (include_checksum) {
         uint8_t checksum = 0;
         for (size_t i = 0; i < len; ++i)
             checksum ^= response[i];
         len = snprintf(response, sizeof(response), "*%u", checksum);
+        len = std::min(len, sizeof(response));
         output.process_bytes((uint8_t*)response, len, nullptr);
```

Comment:
I'm not entirely clear on the USB middleware so this is kind of theoretical. But does sending stuff in three separate `process_bytes` calls slow things down? I was analyzing some data dumps and found many cases where checksums and line breaks showed up in separate USB reads.

Reply:
Yes if the packets are not full (<64B) this slows things down. If throughput becomes a problem for you then combining those three `process_bytes` calls into one could be an effective workaround.

I say workaround because the proper fix would be for the `TreatPacketSinkAsStreamSink` class to coalesce multiple `process_bytes` calls into a single `process_packet` but I wouldn't bother about that right now because the USB code needs an overhaul in general.